scripts/skills/perks/perk_rf_whirling_death.nut

```diff
@@ -144,7 +144,7 @@ this.perk_rf_whirling_death <- ::inherit("scripts/skills/skill", {
 			this.getContainer().setBusy(true);
 			::Time.scheduleEvent(::TimeUnit.Virtual, 100, function ( _perk )
 			{
-				if (target.isAlive())
+				if (user.isAlive() && target.isAlive())
```

Comment:
What do you make of the vanilla check `!actor.isDying()`, which I constantly see alongside the `isAlive` check?
Is it redundant, or just not needed in this particular case?
I dont know the difference/nuance between those two checks

Reply:
I have not looked into it in maximum detail but as far as I looked the `IsDying = true` is set during `actor.kill` function slightly before `IsAlive = false`. So, it seems it is only relevant for certain skill events e.g. `onDeath` where `IsDying` will be true but `IsAlive` will be false.

For our use case here I used to do both `isAlive` and `isDying` checks following some vanilla style, but eventually I realized that the doing `isAlive` is enough. In my opinion the `isDying` check would be redundant and I have so far never encountered any issue in skipping it in skill_container events.

Vanilla also only checks for `target.isAlive()` in the delayed attacks from `cascade_skill.onUse`, so maybe for these cases the `isAlive` check is enough.